src/core/lib/draw-layers.js

```diff
@@ -165,72 +170,88 @@ function drawLayersInViewport(gl, {
 
   // const {x, y, width, height} = deviceRect || [];
 
-  // TODO: Update all layers to use 'picking_uActive' (picking shader module)
-  // and then remove 'renderPickingBuffer' and 'pickingEnabled'.
-  const pickingUniforms = {
-    renderPickingBuffer: drawPickingColors ? 1 : 0,
-    pickingEnabled: drawPickingColors ? 1 : 0
-  };
-
   setParameters(gl, parameters || {});
 
   // render layers in normal colors
   layers.forEach((layer, layerIndex) => {
-    if (layer.isComposite) {
-      renderStats.compositeCount++;
+
+    // Check if we should draw layer
+    let shouldDrawLayer = layer.props.visible;
+    if (drawPickingColors) {
+      shouldDrawLayer = shouldDrawLayer && layer.props.pickable;
+    }
+    if (shouldDrawLayer && layerFilter) {
+      shouldDrawLayer = layerFilter(layer, viewport);
```

Comment:
if layerFilter is a function passed down to <DeckGL/>, can we use that function to set the _visible_ and _pickable_ props?

Reply:
> if layerFilter is a function passed down to , can we use that function to set the visible and pickable props?

Hmm, yes I can see the utility, but how would we do that? Have `layerFilter` return an object with prop overrides?

* Most props cannot be overridden without invoking the layer lifecycle. So those props would be part of a small list of 'special' props that could be overridden in this way?

* We could also add another parameter to `layerFilter` so that it knows if it is picking or rendering?  

If no quick answers, I can make a note and we could take further discussions in the API audit.